### PR TITLE
Feature/lia 47340 serve css

### PR DESCRIPTION
**Pull request by dougschroeder**
@adamayres @meghameshram @savysj @chhamajain @nikhilmodak - changes to SDK to support local skin development for customers via the SDK.

### CODE REVIEW

File: lib/skins.js [hunk 1]
```diff
@@ -19,60 +18,157 @@ var concat = lazyReq('gulp-concat');
...
-        .pipe(sass()({
-          includePaths: includePaths
-        }))
-        .pipe(sourcemaps().write())
-        .pipe(rename()(skin + '.css'))
-        .pipe(gulp.dest(dest))
-        .pipe(lr ? lr() : gutil.noop());
+      withOptions(function (opts) {
+        opts.lr = lr;
+        if (server.useResponsiveConfigsFromServer()) {
```
Comment (dougschroeder) [inline lib/skins.js]: @adamayres I tried to leave the logic you had the same in this script by adding this server.conf.json property check -- if you don't have the useResponsiveConfigsFromServer property set to true, it should behave like it did before. Let me know if you spot anything that doesn't look right or run into problems compiling.

File: test/lib/versioncheck-test.js [hunk 2]
```diff
@@ -9,20 +9,20 @@ var apiHost = 'https://mycommunity.com';
...
+var serverMocks = require('./server-mocks');
 var badErrorRespnse = 'Bad response';
-var successVersion = 15.7;
-var higherVersion = '16.1';
-var higherVersion2 = '15.7.1';
-var higherVersion3 = '15.10';
+var successVersion = 16.2;
+var higherVersion = '16.4';
+var higherVersion2 = '16.3.1';
+var higherVersion3 = '16.3';
```
Comment (meghameshram) [inline test/lib/versioncheck-test.js]: Can we change this test to 16.10? The idea was to test for > 1 digit minor version.

File: lib/plugin-export.js [hunk 3]
```diff
@@ -113,44 +105,64 @@ module.exports = function (gulp, gutil) {
...
-      if (typeof cb !== 'undefined') {
-        cb(new Error(errorMsg), null);
-      } else {
-        throw new Error(errorMsg);
+    function logError(errorMsg) {
+      return putils.logError(gutil, errorMsg);
+    }
+
+    function handleCallback(cb) {
+      if (typeof cb != 'undefined') {
```
Comment (bdharrington7) [inline lib/plugin-export.js]: It would be safer / more correct to check `if (typeof cb === 'function')`

Comment (dougschroeder) [inline lib/plugin-export.js]: Good point -- will change this.

File: test/lib/versioncheck-test.js (hunk 2 above)
Comment (dougschroeder) [inline test/lib/versioncheck-test.js]: will change `higherVersion3` value to `16.10`

**Pull request merged**